client/state/data-layer/http/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import superagent from 'superagent';
+
+/***
+ * Internal dependencies
+ */
+import { extendAction } from 'state/utils';
+import { HTTP_REQUEST } from 'state/action-types';
+import { failureMeta, successMeta } from 'state/data-layer/wpcom-http';
+
+/***
+ * Handler to perorm an http request based on `HTTP_REQUEST` action parameters:
+ * {String} url the url to request
+ * {String} method the method we should use in the request: GET, POST etc.
+ * {Array<{Object}>} headers array of { key: '', value: '' } pairs for the request headers
+ * {Object} body data send as body
+ * {Boolean} withCredentials save cookie set on request
+ * {Action} onSuccess action to dispatch on success with data meta
+ * {Action} onFailure action to dispatch on failure with error meta
+ *
+ * @param {Function} dispatch redux store dispatch
+ * @param {Function} action dispathed action we need to handle
+ * @returns {Promise} promise of the handled request
+ */
+const httpHandler = ( { dispatch }, action ) => {
+	const {
+		url,
+		method,
+		headers,
+		body,
+		withCredentials,
+		onSuccess,
+		onFailure
+	} = action;
+
+	let request = superagent( method, url );
+
+	if ( withCredentials ) {
+		request = request.withCredentials();
+	}
+
+	if ( Array.isArray( headers ) ) {
+		headers.forEach( header => {
+			if ( header.key && header.value ) {
+				request = request.set( header.key, header.value );
+			}
+		} );
+	}
+
+	request = request.accept( 'application/json' );
```

Comment:
I have no idea. They always showcase their API as:
```
request( ... )
.aceept( ... )
.set( ... )
```

So in theory they can just return current request and they can return a new request each time and from the user perspective it would be the same, therefor I prefer to re-assign.